### PR TITLE
#18605 pauses and then unpauses based on a cache invalidation

### PR DESCRIPTION
**Pull request by wezell**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/common/reindex/ReindexThread.java [hunk 1]
```diff
@@ -164,6 +178,19 @@ private BulkProcessor closeBulkProcessor(final BulkProcessor bulkProcessor) thro
     }
 
 
+    private BulkProcessor finalizeReIndex(BulkProcessor bulkProcessor) throws InterruptedException, LanguageException, DotDataException, SQLException {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/common/reindex/ReindexThread.java:181]: Codacy found an issue: [Avoid reassigning parameters such as 'bulkProcessor'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6373630)

File: dotCMS/src/main/java/com/dotmarketing/common/reindex/ReindexThread.java [hunk 2]
```diff
@@ -164,6 +174,19 @@ private BulkProcessor closeBulkProcessor(final BulkProcessor bulkProcessor) thro
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/common/reindex/ReindexThread.java:177]: Codacy found an issue: [Avoid reassigning parameters such as 'bulkProcessor'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6373630)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/common/reindex/ReindexThread.java:177]: Codacy found an issue: [Avoid reassigning parameters such as 'bulkProcessor'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6373630)

**Review by freddyucv — COMMENTED** [review]
is it possible do any test here?

File: dotCMS/src/main/java/com/dotmarketing/common/reindex/ReindexThread.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/common/reindex/ReindexThread.java:177]: Codacy found an issue: [Avoid reassigning parameters such as 'bulkProcessor'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6373630)

**Comment by nollymar** [comment]
> is it possible do any test here?

Done

File: dotCMS/src/integration-test/java/com/dotmarketing/common/reindex/ReindexAPITest.java [hunk 3]
```diff
@@ -123,42 +123,47 @@ public void test_highestpriority_reindex_vs_normal_reindex() throws DotDataExcep
...
-        }
-
-        final ReindexQueueAPI reindexQueueAPI = APILocator.getReindexQueueAPI();
-
-
-
-        Map<String, ReindexEntry> reindexEntries = reindexQueueAPI.findContentToReindex(numberToTest);
-
-        assertTrue("should have " + numberToTest + " to reindex, only got" +  reindexEntries.size(),  reindexEntries.size()== numberToTest);
+        Config.setProperty("ALLOW_MANUAL_REINDEX_UNPAUSE", true);
```
Comment (erickgonzalez) [inline dotCMS/src/integration-test/java/com/dotmarketing/common/reindex/ReindexAPITest.java:126]: Doc Test

**Review by erickgonzalez — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

**Pull request merged**